govcd/edgegateway.go

```diff
@@ -1076,3 +1076,88 @@ func (eGW *EdgeGateway) buildProxiedEdgeEndpointURL(optionalSuffix string) (stri
 }
+
+// GetLoadBalancer retrieves load balancer configuration of `&types.LoadBalancer` and can be used
+// to access global configuration options. These are 4 fields only:
+// LoadBalancer.Enabled, LoadBalancer.AccelerationEnabled, LoadBalancer.Logging.Enable,
+// LoadBalancer.Logging.LogLevel
+func (egw *EdgeGateway) GetLoadBalancer() (*types.LoadBalancer, error) {
+	if !egw.HasAdvancedNetworking() {
+		return nil, fmt.Errorf("only advanced edge gateway supports load balancing")
+	}
+
+	httpPath, err := egw.buildProxiedEdgeEndpointURL(types.LBConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
+	}
+
+	loadBalancerConfig := &types.LoadBalancer{}
+	_, err = egw.client.ExecuteRequest(httpPath, http.MethodGet, types.AnyXMLMime,
+		"unable to read load balancer configuration: %s", nil, loadBalancerConfig)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return loadBalancerConfig, nil
+}
+
+// UpdateLoadBalancer allows to update global load balancer configuration.
+// These are 4 fields only: LoadBalancer.Enabled, LoadBalancer.AccelerationEnabled,
+// LoadBalancer.Logging.Enable, LoadBalancer.Logging.LogLevel. They are represented in load balancer
+// global configuration tab in the UI.
+// All other fields are ignored and sent as they are in order to prevent load balancer configuration
+// corruption
+func (egw *EdgeGateway) UpdateLoadBalancer(lb *types.LoadBalancer) (*types.LoadBalancer, error) {
+	if !egw.HasAdvancedNetworking() {
+		return nil, fmt.Errorf("only advanced edge gateway supports load balancing")
+	}
+
+	if err := validateUpdateLoadBalancer(lb); err != nil {
+		return nil, err
+	}
+	// Retrieve load balancer to work on latest configuration
+	currentLb, err := egw.GetLoadBalancer()
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve load balancer before update: %s", err)
+	}
+
+	// Modify only the global configuration settings
+	currentLb.Enabled = lb.Enabled
+	currentLb.AccelerationEnabled = lb.AccelerationEnabled
+	currentLb.Logging = lb.Logging
+	// Ommit the version as it is updated automatically with each put
+	currentLb.Version = ""
+
+	// Push updated configuration
+	httpPath, err := egw.buildProxiedEdgeEndpointURL(types.LBConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
+	}
+	_, err = egw.client.ExecuteRequestWithCustomError(httpPath, http.MethodPut, types.AnyXMLMime,
+		"error while updating load balancer application rule : %s", currentLb, &types.NSXError{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Retrieve configuration after update
+	updatedLb, err := egw.GetLoadBalancer()
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve load balancer after update: %s", err)
+	}
+
+	return updatedLb, nil
+}
+
+// validateUpdateLoadBalancer validates mandatory fields for global load balancer configuration
+// settings
+func validateUpdateLoadBalancer(lb *types.LoadBalancer) error {
```

Comment:
Kind'a strange. I thouhgh LoadBalancer.Enabled, LoadBalancer.AccelerationEnabled are main thing's

Reply:
Yeah, but these are `bool`. They have a default value (of false), I thought about using `*bool` to evaluate `nil` conditions, but then the constructs become quite ugly as you must use an intermediate var as you cannot do something like `&false`

Reply:
Yep, in such cases then we expose values which implicitly set default values and we request from user to have them, we kinda help them and also hide what they may explicitly set. I think topic for discussion for overall API.

Reply:
Actually initially when you get the load balancer it is disabled so the default boolean value `false` plays quite fair here.

Reply:
ok then, in this case.